crates/polars-core/src/frame/hash_join/args.rs

```diff
@@ -131,40 +131,62 @@ impl JoinValidation {
         Ok(())
     }
 
-    pub(super) fn validate_probe(
+    pub(super) fn validate_probe<'a, F, I, T>(
```

Comment:
I am not super happy with the amount of code this generates, as this will be compiled for every `T`. Was `s_unique() == s.len()` wrong?

Reply:
> I am not super happy with the amount of code this generates, as this will be compiled for every `T`. Was `s_unique() == s.len()` wrong?

It not wrong, I will revert it. 

The reason I change this is that the this mod checked and swapped left and right multiple times, which makes it really confusing. But after looking into this, I should not change this function.